### PR TITLE
Creating and getting a project's taxonomy

### LINKED ISSUE #171: Create a taxonomy for a classification project
# 🙋 feature request

### PR DESCRIPTION
**Pull request by juliamcclellan**
Fix #171

### CODE REVIEW

File: tamr_unify_client/models/project/categorization.py [hunk 1]
```diff
@@ -15,4 +16,30 @@ def model(self):
...
+        alias = self.api_path + "/taxonomy"
+        resource_json = self.client.post(
+            alias, json={"name":  name}
+        ).successful().json()
+        return Taxonomy.from_json(self.client, resource_json, alias)
+
+    def taxonomy(self):
+        """Retrieves the Taxonomy associated with Categorization project.
+
+        If a taxonomy is not already associated with this project, call create_taxonomy() first.
```
Comment (charlottemoremen) [inline tamr_unify_client/models/project/categorization.py]: what happens if you call taxonomy on a project that doesn't have one associated with it? should it raise an error and say something along the lines of "you need to create a taxonomy for this project" ?

Comment (juliamcclellan) [reply tamr_unify_client/models/project/categorization.py]: There's a 404 error

File: tamr_unify_client/models/project/categorization.py [hunk 2]
```diff
@@ -15,4 +16,28 @@ def model(self):
         alias = self.api_path + "/categorizations/model"
         return MachineLearningModel(self.client, None, alias)
 
+    def create_taxonomy(self, name=""):
```
Comment (nbateshaus) [inline tamr_unify_client/models/project/categorization.py]: This should take a `creation_spec` even though it just contains a `name` right now.
As the API adds features to taxonomies, we don't want existing code to break.

File: tamr_unify_client/models/project/categorization.py [hunk 3]
```diff
@@ -15,4 +16,28 @@ def model(self):
...
+        """Creates a Taxonomy for this Categorization project.
+
+        A taxonomy cannot already be associated with this project.
+
+        :returns: The new Taxonomy
+        :rtype: :class:`~tamr_unify_client.models.taxonomy.resource.Taxonomy
+        """
+        alias = self.api_path + "/taxonomy"
+        resource_json = self.client.post(alias, json={"name": name}).successful().json()
```
Comment (nbateshaus) [inline tamr_unify_client/models/project/categorization.py]: We should pass the `creation_spec` as the `json=` parameter.

File: tests/unit/test_categorization.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        taxonomy_url = f"http://localhost:9100/api/versioned/v1/projects/1/taxonomy"
+        responses.add(responses.GET, project_url, json=self._project_json)
+        responses.add(responses.POST, taxonomy_url, json=self._taxonomy_json)
+
+        project = self.unify.projects.by_resource_id("1").as_categorization()
+        u = project.create_taxonomy("Test Taxonomy")
+
+        responses.add(responses.GET, taxonomy_url, json=self._taxonomy_json)
+        t = project.taxonomy()
+        assert print(u) == print(t)
```
Comment (nbateshaus) [inline tests/unit/test_categorization.py]: use `self.assertEqual()` rather than `assert` - it is better :-)

**Pull request merged**